src/prototype_creep_harvest.js

```diff
@@ -59,9 +59,9 @@ Creep.prototype.spawnCarry = function() {
   let carryCapacity = config.carry.sizes[foundKey][1] * CARRY_CAPACITY;
 
   const workParts = this.body.filter(part => part.type === WORK).length;
-
-  let waitTime = carryCapacity / (HARVEST_POWER * workParts);
-
+  let distance = Game.map.getRoomLinearDistance(this.room.name, this.memory.base);
+  let waitTime = Math.floor(1500 * carryCapacity * distance / (2 * this.memory.timeToTravel * (HARVEST_POWER * workParts))) ;
```

Comment:
Time between carrys should be proportional to % of energy the carry will carry in his life as : 
Energy harvested in sourcer life :  
`1500 * (HARVEST_POWER * workParts) = A`
Energy carried by carry : 
`1500 * carryCapacity / 2 * timetotravel = B`
`B/A = carryCapacity /( 2 * timetotravel * harvestpower * workParts) `
This result should be interpreted as spawn a creep each 1500 ticks is B/A = 1. One each 750 ticks if it's 1/2.
then we just have to multiply it by 1500 and floor all for have a rounded value.  
The distance coeff is for take care of surpopulation in parents rooms, then far rooms will call less carry.

Reply:
take `resourceAtPosition` into account would be better.